src/compiler/scanner.ts

```diff
@@ -2384,15 +2444,979 @@ export function createScanner(languageVersion: ScriptTarget, skipTrivia: boolean
+                                }
+                            }
+                            else if (!min) {
+                                if (unicodeMode) {
+                                    error(Diagnostics.Unexpected_0_Did_you_mean_to_escape_it_with_backslash, start, 1, String.fromCharCode(ch));
+                                }
+                                isPreviousTermQuantifiable = true;
+                                break;
+                            }
+                            scanExpectedChar(CharacterCodes.closeBrace);
+                            pos--;
+                        // falls through
+                        case CharacterCodes.asterisk:
+                        case CharacterCodes.plus:
+                        case CharacterCodes.question:
+                            pos++;
+                            if (text.charCodeAt(pos) === CharacterCodes.question) {
+                                // Non-greedy
+                                pos++;
+                            }
+                            if (!isPreviousTermQuantifiable) {
+                                error(Diagnostics.There_is_nothing_available_for_repetition, start, pos - start);
+                            }
+                            isPreviousTermQuantifiable = false;
+                            break;
+                        case CharacterCodes.dot:
+                            pos++;
+                            isPreviousTermQuantifiable = true;
+                            break;
+                        case CharacterCodes.openBracket:
+                            pos++;
+                            if (unicodeSetsMode) {
+                                scanClassSetExpression();
+                            }
+                            else {
+                                scanClassRanges();
+                            }
+                            scanExpectedChar(CharacterCodes.closeBracket);
+                            isPreviousTermQuantifiable = true;
+                            break;
+                        case CharacterCodes.closeParen:
+                            if (isInGroup) {
+                                return;
+                            }
+                        // falls through
+                        case CharacterCodes.closeBracket:
+                        case CharacterCodes.closeBrace:
+                            if (isUnterminated && !isInGroup) {
+                                // Assume what starting from the character to be outside of the regex
+                                return;
+                            }
+                            if (unicodeMode) {
+                                error(Diagnostics.Unexpected_0_Did_you_mean_to_escape_it_with_backslash, pos, 1, String.fromCharCode(ch));
+                            }
+                            pos++;
+                            isPreviousTermQuantifiable = true;
+                            break;

```

Comment:
Don't you need to make sure that these all match up with the matching opener?

Reply:
`scanAlternative` itself suggests that the opener is an open parenthesis.